app/controllers/UploadYourFilesReceiptController.scala

```diff
@@ -39,20 +41,29 @@ class UploadYourFilesReceiptController @Inject()(
   cdsFileUploadConnector: CdsFileUploadConnector,
   metrics: SfusMetrics,
   uploadYourFilesReceipt: upload_your_files_receipt,
-  answersConnector: AnswersService
+  uploadYourFilesConfirmation: upload_your_files_confirmation,
+  answersConnector: AnswersService,
+  secureMessagingConfig: SecureMessagingConfig
 )(implicit mcc: MessagesControllerComponents, ec: ExecutionContext)
     extends FrontendController(mcc) with I18nSupport {
 
   def onPageLoad(): Action[AnyContent] = (authenticate andThen verifiedEmail).async { implicit req =>
     answersConnector.findByEori(req.eori).flatMap { maybeUserAnswers =>
+      answersConnector.removeByEori(req.eori)
```

Comment:
Not "really" safe this op (removing after retrieve I mean, as mongo 3.6 is still non-transactional), but maybe it is acceptable here, I do not know. I do not remember exactly, but if I'm not wrong you could use a bulk op with find and remove with execute as last step, in the repository of course, not here or in the service.

Reply:
After our discussion have agreed that this isn't an issue for this scenario so leaving as is.